internal/receiver/discoveryreceiver/endpoint_tracker.go

```diff
@@ -368,31 +356,50 @@ func extractIdentifyingAttrs(from pcommon.Map, to pcommon.Map) {
 	}
 }
 
-func deduceServiceName(attrs pcommon.Map) string {
-	if val, ok := attrs.Get(string(conventions.ServiceNameKey)); ok {
-		return val.AsString()
+func extractServiceName(endpointType observer.EndpointType, endpointEnv observer.EndpointEnv) string {
+	labels, labelsFound := endpointEnv["labels"].(map[string]string)
+	podEnv, podEnvFound := endpointEnv["pod"].(observer.EndpointEnv)
```

Comment:
Is my understanding correct that we'll have this `pod` map available with labels, etc for `Port` type observer?

Reply:
Right